tests/unit_tests/ensemble_evaluator/test_sync_ws_duplexer.py

```diff
@@ -1,201 +0,0 @@
-
-
-def test_failed_send(unused_tcp_port, ws):
-    async def handler(websocket, path):
-        await websocket.recv()
-
-    ws("localhost", unused_tcp_port, handler)
-    duplexer = SyncWebsocketDuplexer(
-        f"ws://localhost:{unused_tcp_port}",
-        f"ws://localhost:{unused_tcp_port}",
-        None,
-        None,
-    )
-    with patch("websockets.WebSocketCommonProtocol.send") as send:
-        send.side_effect = OSError("expected OSError")
-        with pytest.raises(OSError, match="expected OSError"):
-            duplexer.send("hello")
-
-
-def test_unexpected_close(unused_tcp_port, ws):
-    async def handler(websocket, path):
-        await websocket.close()
-
-    ws("localhost", unused_tcp_port, handler)
-    with ExitStack() as stack:
-        duplexer = SyncWebsocketDuplexer(
-            f"ws://localhost:{unused_tcp_port}",
-            f"ws://localhost:{unused_tcp_port}",
-            None,
-            None,
-        )
-        stack.callback(duplexer.stop)
-        with pytest.raises(ConnectionClosedOK):
-            next(duplexer.receive())
-
-
-def test_receive(unused_tcp_port, ws):
-    async def handler(websocket, path):
-        await websocket.send("Hello World")
-
-    ws("localhost", unused_tcp_port, handler)
-    with ExitStack() as stack:
-        duplexer = SyncWebsocketDuplexer(
-            f"ws://localhost:{unused_tcp_port}",
-            f"ws://localhost:{unused_tcp_port}",
-            None,
-            None,
-        )
-        stack.callback(duplexer.stop)
-        assert next(duplexer.receive()) == "Hello World"
-
-
-def test_echo(unused_tcp_port, ws):
-    async def handler(websocket, path):
-        msg = await websocket.recv()
-        await websocket.send(msg)
-
-    ws("localhost", unused_tcp_port, handler)
-    with ExitStack() as stack:
-        duplexer = SyncWebsocketDuplexer(
-            f"ws://localhost:{unused_tcp_port}",
-            f"ws://localhost:{unused_tcp_port}",
-            None,
-            None,
-        )
-        stack.callback(duplexer.stop)
-
-        duplexer.send("Hello World")
-        assert next(duplexer.receive()) == "Hello World"
-
-
-def test_generator(unused_tcp_port, ws):
-    async def handler(websocket, path):
-        await websocket.send("one")
-        await websocket.send("two")
-        await websocket.send("three")
-        await websocket.send("four")
-
-    ws("localhost", unused_tcp_port, handler)
-    with ExitStack() as stack:
-        duplexer = SyncWebsocketDuplexer(
-            f"ws://localhost:{unused_tcp_port}",
-            f"ws://localhost:{unused_tcp_port}",
-            None,
-            None,
-        )
-        stack.callback(duplexer.stop)
-
-        expected = ["one", "two", "three"]
-        for msg in duplexer.receive():
-            assert msg == expected.pop(0)
-
-            # Cause a GeneratorExit
-            if len(expected) == 1:
-                break
-
-
-def test_secure_echo(ws):
```

Comment:
Has these tests been evaluated whether they are relevant to port to the new monitor?

Reply:
So, I have cherry-picked and refactored some tests, but not really sure about their usability. You are free to comment :)